DOCSUP-3123: Edited the ReplicatedMergeTree and Polygon dictionaries (#16110)
ClickHouse/ClickHouse: docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-polygon.md

---
toc_priority: 46
toc_title: Polygon Dictionaries With Grids 
---

# Polygon dictionaries {#polygon-dictionaries}

> Polygon dictionaries allow you to efficiently search for the polygon that these points fall into amongst a difference of polygons.

Reviewer: ```suggestion
Polygon dictionaries allow you to efficiently search for the polygon containing specified points.
```

For example: defining a city area by geographical coordinates.

Example configuration:

``` xml
<dictionary>
    <structure>
        <key>
            <name>key</name>
            <type>Array(Array(Array(Array(Float64))))</type>
        </key>

        <attribute>
            <name>name</name>
            <type>String</type>
            <null_value></null_value>
        </attribute>

        <attribute>
            <name>value</name>
            <type>UInt64</type>
            <null_value>0</null_value>
        </attribute>

    </structure>

    <layout>
        <polygon />
    </layout>

</dictionary>
```

Tne corresponding [DDL-query](../../../sql-reference/statements/create/dictionary.md#create-dictionary-query):
``` sql
CREATE DICTIONARY polygon_dict_name (
    key Array(Array(Array(Array(Float64)))),
    name String,
    value UInt64
)
PRIMARY KEY key
LAYOUT(POLYGON())
...
```

When configuring the polygon dictionary, the key must have one of two types:
-   A simple polygon. It is an array of points.

> -   MultiPolygon. It is an array of polygons. Each polygon is a two-dimensional array of points. The first element of this array is the outer boundary of the polygon, and subsequent elements specify holes in from it.

Reviewer: ```suggestion
-   MultiPolygon. It is an array of polygons. Each polygon is a two-dimensional array of points. The first element of this array is the outer boundary of the polygon, and subsequent elements specify areas to be excluded from it.
```
